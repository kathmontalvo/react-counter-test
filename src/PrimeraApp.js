import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitle = 'Soy un subtitle' }) => {

  
  return (
    <div>
      <h1>{ saludo }!!</h1>
      <p>{ subtitle }</p>
      {/* <pre> {JSON.stringify(obj, null, 3)} </pre> */}
    </div>
  )
}

// PROPTYPES
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
  subtitle: 'Soy un subtitle'
}

export default PrimeraApp;