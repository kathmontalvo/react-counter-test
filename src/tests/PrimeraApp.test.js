import React from 'react';
// import { render } from "@testing-library/react";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import PrimeraApp from "../PrimeraApp";

describe('<Pruebas en PrimeraApp />', () => {
  
  // test('deberia mostrar el msj Hola, soy Boddy perrito ', () => {
  //   const saludo = 'Hola, soy Boddy perrito';

  //   const { getByText } = render( <PrimeraApp saludo={saludo} /> );

  //   expect(getByText(saludo)).toBeInTheDocument();

  // })

  test('debe mostrar <PrimeraApp /> correctamente', () => {
    
    const saludo = 'Hola, soy Boddy perrito';
    const wrapper = shallow(<PrimeraApp saludo= {saludo} />);
    
    //esperaria q se renderice de la manera correcta
    expect(wrapper).toMatchSnapshot();

  })

  test('debe mostrar el subtitulo q se envia por props', () => {

    const saludo = 'Hola, soy Boddy perrito';
    const subtitle = 'Un perrito muy bonito';
    
    const wrapper = shallow(
      <PrimeraApp 
        saludo= {saludo} 
        subtitle = {subtitle}
        />
    );
    // lo siguiente funciona como un querySelector
    const textoParrafo = wrapper.find('p').text();
    expect(textoParrafo).toBe(subtitle);
    
    
  })
  
  
  
})
