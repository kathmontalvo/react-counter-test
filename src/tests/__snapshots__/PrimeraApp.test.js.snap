// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pruebas en PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<div>
  <h1>
    Hola, soy Boddy perrito
    !!
  </h1>
  <p>
    Soy un subtitle
  </p>
</div>
`;
